[Merged by Bors] - refactor(Algebra/Module): Move `Module.ofCore` to a MinimalAxioms file, and rename it `ofMinimalAxioms`

martincmartin opened the pull request:
This makes it consistent with Ring, Field and Group.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

martincmartin commented:
Should I rename the structure `Module.Core` to `Module.MinimalAxioms`?  Or just get rid of it, to be consistent with the other `ofMinimalAxioms` functions?

eric-wieser commented on Mathlib/Algebra/Module/MinimalAxioms.lean:
```diff
@@ -0,0 +1,49 @@
+import Mathlib.Algebra.Module.Basic
+
+/-!
+# Minimal Axioms for a Module.
```
```suggestion
# Minimal Axioms for a Module
```

eric-wieser commented on Mathlib/Algebra/Module/MinimalAxioms.lean:
```diff
@@ -0,0 +1,49 @@
+# Minimal Axioms for a Module.
+
+This file defines a constructor to define a `Module` structure on a Type with an
+AddCommGroup, while proving a minimum number of equalities.
```
```suggestion
`AddCommGroup`, while proving a minimum number of equalities.
```

eric-wieser reviewed (approved):
LGTM, thanks!

Can you update the PR title to reflect the [commit conventions](https://leanprover-community.github.io/contribute/commit.html)?

eric-wieser commented:
> Or just get rid of it, to be consistent with the other `ofMinimalAxioms` functions?

This sounds like a good idea to me. The original design is vestigial from the refactor in https://github.com/leanprover-community/mathlib/pull/456, so I don't think we need to be attached to it.

martincmartin commented:
@eric-wieser I believe I've addressed all your comments.  Please let me know if you'd like me to change anything else before merging.  And thanks for reviewing and your helpful comments!

eric-wieser commented on Mathlib/Algebra/Module/MinimalAxioms.lean:
```diff
@@ -0,0 +1,40 @@
+universe u v
+
+/-- Define a `Module` structure on a Type by proving a minimized set of axioms. -/
+def Module.ofMinimalAxioms {R : Type u} {M : Type v} [Semiring R] [AddCommGroup M] [SMul R M]
```
Using `reducible` instead for consistency is also fine.

eric-wieser replied:
```suggestion
@[reducible]
def Module.ofMinimalAxioms {R : Type u} {M : Type v} [Semiring R] [AddCommGroup M] [SMul R M]
```

eric-wieser commented:
bors merge

Thanks!

eric-wieser commented:
bors merge
